turbinia/lib/recipe_helpers.py

```diff
@@ -0,0 +1,135 @@
+      log.error(message)
+      return False
+    except IOError as exception:
+      log.error(
+          'Failed to read recipe file {0:s}: {1!s}'.format(
+              recipe_file, exception))
+      validate_recipe(recipe_dict)
+      return False
+    return recipe_dict
+
+
+def validate_globals_recipe(proposed_globals_recipe):
+  """Ensures globals recipe is valid for further processing."""
+
+  for item in DEFAULT_GLOBALS_RECIPE:
+    if item not in proposed_globals_recipe:
+      proposed_globals_recipe[item] = DEFAULT_GLOBALS_RECIPE[item]
+
+  filter_patterns_file = proposed_globals_recipe.get(
+      'filter_patterns_file', None)
+  yara_rules_file = proposed_globals_recipe.get('yara_rules_file', None)
+  if filter_patterns_file:
+    proposed_globals_recipe['filter_patterns'] = file_to_list(
+        filter_patterns_file)
+  if yara_rules_file:
+    proposed_globals_recipe['yara_rules'] = file_to_str(yara_rules_file)
+  diff = set(DEFAULT_GLOBALS_RECIPE) - set(proposed_globals_recipe)
+  if diff:
+    log.error('Unknown key {0:s} found on globals recipe item'.format(diff))
+    return False
+
+  if any(i in proposed_globals_recipe['jobs_denylist']
+         for i in proposed_globals_recipe['jobs_allowlist']):
+    log.error('No jobs can be simultaneously in the allow and deny lists')
+    return False
+  return True
+
+
+def validate_task_recipe(proposed_recipe, task_config):
+  """Ensure only allowed parameters are present a given task recipe."""
+  return proposed_recipe.items() <= task_config.items()
+
+
+def validate_recipe(recipe_dict):
+  """Validate the 'task_recipes' dict supplied by the request recipe."""
+  tasks_with_recipe = []
+  valid_recipe = True
+  #If not globals task recipe is specified create one.
+  if 'globals' not in recipe_dict:
+    recipe_dict['globals'] = copy.deepcopy(DEFAULT_RECIPE)
+    log.warning(
+        'No globals recipe specified, all recipes should include a globals entry, the default values will be used'
+    )
+
+  for recipe_item, recipe_item_contents in recipe_dict.items():
+    if recipe_item in tasks_with_recipe:
+      log.error(
+          'Two recipe items with the same name {0:s} have been found.'
+          'If you wish to specify several task runs of the same tool,'
+          'please include them in separate recipes.'.format(recipe_item))
+      valid_recipe = False
+    if 'task' not in recipe_item_contents:
+      if recipe_item != 'globals':
+        log.error(
+            'Recipe item {0:s} has no "task" key. All recipe items must have a "task" key indicating the TurbiniaTask'
+            ' to which it relates.'.format(recipe_item))
+        valid_recipe = False
+      else:
+        if not validate_globals_recipe(recipe_item_contents):
+          log.error('Invalid globals recipe.')
+          valid_recipe = False
+    else:
+      proposed_task = recipe_item_contents['task']
+      if proposed_task not in [v.__name__ for v in TASK_MAP.values()]:
```

Comment:
I think the following would be simpler since the keys are also the task names in lower case:
```suggestion
      if lower(proposed_task) not in TASK_MAP:
```